### PR TITLE
chore(web): add support for layer name editing in beta

### PR DESCRIPTION
**Pull request by pyshx**
# Overview
This PR adds support for layer name editing on double click on the title of the layer in Map/LeftPanel.

### CODE REVIEW

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx [hunk 1]
```diff
@@ -57,30 +63,74 @@ const Layers: React.FC<LayersProps> = ({
...
+  id: string;
   layerTitle: string;
   onDelete: () => void;
   onSelect: () => void;
+  onLayerNameUpdate: (inp: LayerNameUpdateProps) => void;
 };
 
-const LayerItem: React.FC<LayerItemProps> = ({ layerTitle, onDelete, onSelect }) => {
-  const [isMenuOpen, setMenuOpen] = React.useState(false);
+const LayerItem = ({ id, layerTitle, onDelete, onSelect, onLayerNameUpdate }: LayerItemProps) => {
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx:85]: Please extract this out. Can just move it in to `Layers/LayerItem.tsx` 🙏🏻

File: web/src/beta/components/fields/TextInput/index.tsx [hunk 2]
```diff
@@ -38,15 +48,17 @@ const TextInput: React.FC<Props> = ({ name, description, value, timeout = 1000,
   const handleBlur = useCallback(() => {
     if (timeoutRef.current) clearTimeout(timeoutRef.current);
     onChange?.(currentValue);
-  }, [currentValue, onChange]);
+    onBlur?.();
+  }, [currentValue, onChange, onBlur]);
 
   const handleKeyDown = useCallback(
```
Comment (KaWaite) [inline web/src/beta/components/fields/TextInput/index.tsx:54]: handleKeyUp

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx [hunk 3]
```diff
@@ -57,30 +63,74 @@ const Layers: React.FC<LayersProps> = ({
...
+        handleTitleSubmit();
+      }
+    },
+    [handleTitleSubmit],
+  );
+
+  const handleBlur = () => {
+    setIsEditing(false);
+    setEditedTitle(prevTitle);
+  };
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx:119]: For other inputs clicking away will set the new value (aka save the value). Can you make handleblur call onLayerNameUpdate? 🙏🏻

Comment (KaWaite) [reply web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx:119]: Also useCallback

File: web/src/beta/components/fields/TextInput/index.tsx (hunk 2 above)
Comment (pyshx) [reply web/src/beta/components/fields/TextInput/index.tsx]: Done.

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx (hunk 3 above)
Comment (pyshx) [reply web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx]: Done.

**Pull request merged**